fast/mediastream/video-srcObject-set-twice.html is flaky

youennf opened the pull request:
#### 13fa2acdd4842002cebe0ba83ca272cb4213f391
<pre>
fast/mediastream/video-srcObject-set-twice.html is flaky
<a href="https://bugs.webkit.org/show_bug.cgi?id=258851">https://bugs.webkit.org/show_bug.cgi?id=258851</a>
rdar://111738593

Reviewed by Philippe Normand.

When taking a snapshot, we sometimes do not see the video element.
This might be related to rendering taking some time.
We replace the 500 ms timer by calling testRunner.takeViewPortSnapshot and checking the result before calling notifyDone.
If the result is not as expected, we wait for the next frame to retry.
We add console logs in case we need to do some additional debugging from bot results.

* LayoutTests/fast/mediastream/video-srcObject-set-twice.html:

Canonical link: <a href="https://commits.webkit.org/265767@main">https://commits.webkit.org/265767@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/830a80c6cc661a3c74688b2f6731d2c1218b3a2d

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/11805 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/12004 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/12374 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/13449 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/11244 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/14393 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/11985 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/14103 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/11969 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/12799 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/10006 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/13871 "Built successfully") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/39/builds/10091 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/10718 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/17848 "Passed tests") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/11169 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/10873 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/14043 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/11281 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/17/builds/9320 "Passed tests") | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/10455 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/10597 "Passed tests") | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/2850 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/14738 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/11137 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

philn commented on LayoutTests/fast/mediastream/video-srcObject-set-twice.html:
```diff
@@ -1,12 +1,88 @@
+
+async function validateSnapshot()
+{
+console.log("validateSnapshot-1 ");
```
These a debug print left-overs I guess? Maybe remove them

youennf replied:
They will go in stderr.txt, which might be useful if the test keeps being flakey in bots.

youennf committed:
fast/mediastream/video-srcObject-set-twice.html is flaky
https://bugs.webkit.org/show_bug.cgi?id=258851
rdar://111738593

Reviewed by Philippe Normand.

When taking a snapshot, we sometimes do not see the video element.
This might be related to rendering taking some time.
We replace the 500 ms timer by calling testRunner.takeViewPortSnapshot and checking the result before calling notifyDone.
If the result is not as expected, we wait for the next frame to retry.
We add console logs in case we need to do some additional debugging from bot results.

* LayoutTests/fast/mediastream/video-srcObject-set-twice.html:

Canonical link: https://commits.webkit.org/265767@main
